scipy/interpolate/interpolate.py

```diff
@@ -987,12 +1003,54 @@ def integrate(self, a, b, extrapolate=None):
             a, b = b, a
             sign = -1
 
-        # Compute the integral
         range_int = np.empty((prod(self.c.shape[2:]),), dtype=self.c.dtype)
         self._ensure_c_contiguous()
-        _ppoly.integrate(self.c.reshape(self.c.shape[0], self.c.shape[1], -1),
-                         self.x, a, b, bool(extrapolate),
-                         out=range_int)
+
+        # Compute the integral.
+        if extrapolate == 'periodic':
+            # Split the integral into the part over period (can be several
+            # of them) and the remaining part.
+
+            xs, xe = self.x[0], self.x[-1]
+            period = xe - xs
+            interval = b - a
+            n_periods = int(interval / period)
+
+            if n_periods > 0:
+                _ppoly.integrate(
+                    self.c.reshape(self.c.shape[0], self.c.shape[1], -1),
+                    self.x, xs, xe, False, out=range_int)
+                range_int *= n_periods
+                a += n_periods * period
+            else:
+                range_int.fill(0)
+
+            # Map a and b to [xs, xe].
+            a = xs + (a - xs) % period
+            b = xs + (b - xs) % period
```

Comment:
I have a feeling that `%` and `int(interval / period)` are not necessarily always consistent with each other due to rounding error. 
Can the computation be recast in terms of `divmod` only?